### PR TITLE
Fix docker upgrades and volume interplay.

### PR DESCRIPTION
**Pull request by postwait**
If /opt/noit/prod/etc is a volume, we get the "original" installs
copy in there and it will never upgrade.  To work around that we
stash a copy of the deb used to install these files in /opt/napp/package
and if it doesn't match the version last used in the volume we force
a reinstall (which don't touch the user configs) and mark it as done.
This should happen once per upgrade.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -9,8 +9,11 @@ RUN chmod 755 /bin/console && \
     apt-get install -y circonus-field-broker && \
     apt-get install -y --allow-unauthenticated circonus-field-broker-crashreporter && \
     apt-get install -y telnet && \
+    mkdir /opt/napp/package && \
+    /opt/circonus/bin/curl -o /opt/napp/package/circonus-field-broker-core-stock.deb http://pilot.circonus.net/ubuntu/pool/main/c/circonus-field-broker-core/`dpkg -l circonus-field-broker-core | awk '/broker-core/{print $2"-"$3"xenial_"$4".deb"}'` && \
+    /usr/bin/dpkg -l circonus-field-broker-core | \
+      awk '/broker-core/{print $2"-"$3}' > /opt/noit/prod/etc/docker-core-version && \
     rm -rf /var/lib/apt/lists/*
-RUN echo "MANAGED_CORONER=1" >> /opt/noit/prod/etc/noit.env
```
Comment (esproul) [inline docker/Dockerfile:13]: why is this removed?

Comment (postwait) [reply docker/Dockerfile:13]: the broker-start script auto-detects running from within docker and will set this variable.

File: appliance-root/opt/napp/bin/broker-start [hunk 2]
```diff
@@ -1,11 +1,35 @@
 #!/bin/sh
 
+###
+# In docker we want so support /opt/noit/prod/etc as a volume
+# However, this means when we roll new docker images and people upgrade
+# the will not get the updated package config files.  While noone has
```
Comment (esproul) [inline appliance-root/opt/napp/bin/broker-start:6]:
```suggestion
# they will not get the updated package config files.  While no one has
```

File: appliance-root/opt/napp/bin/broker-start [hunk 3]
```diff
@@ -1,11 +1,35 @@
...
+# ever said anything nice about apt/dpkg, at least it knows which config
+# files are user-editable...  So, for the docker image we drop a copy of
+# the circonus-field-broker-core package we installed into
+#   /opt/napp/package/circonus-field-broker-core-stock.deb
+#
+# This might not be the same one that was installed when these volumes
+# were created, so we just go ahead and for install it again and it will
```
Comment (esproul) [inline appliance-root/opt/napp/bin/broker-start:13]:
```suggestion
# were created, so we just go ahead and force-install it again and it will
```

File: docker/Dockerfile [hunk 4]
```diff
@@ -9,8 +9,11 @@ RUN chmod 755 /bin/console && \
     apt-get install -y circonus-field-broker && \
     apt-get install -y --allow-unauthenticated circonus-field-broker-crashreporter && \
     apt-get install -y telnet && \
+    mkdir /opt/napp/package && \
+    /opt/circonus/bin/curl -o /opt/napp/package/circonus-field-broker-core-stock.deb http://pilot.circonus.net/ubuntu/pool/main/c/circonus-field-broker-core/`dpkg -l circonus-field-broker-core | awk '/broker-core/{print $2"-"$3"xenial_"$4".deb"}'` && \
```
Comment (esproul) [inline docker/Dockerfile:13]: We might want to encode the LSB release codename here too, thinking ahead to 20.04/focal support.

Comment (postwait) [reply docker/Dockerfile:13]: I don't think that necessary.  This stuff only makes sense for config file changes, which are only depend on version (not release, or platform, or even architecture)
